src/sentry_plugins/github/plugin.py

```diff
@@ -35,6 +38,44 @@
 )
 
 
+def backfill_commit(repo, commit):
+    author_email = commit['commit']['author']['email']
+    if '@' not in author_email:
+        author_email = u'{}@localhost'.format(
+            author_email[:65],
+        )
+
+    # TODO(dcramer): we need to deal with bad values here, but since
+    # its optional, lets just throw it out for now
+    if len(author_email) > 75:
+        author = None
+    else:
+        author = CommitAuthor.objects.get_or_create(
+            organization_id=repo.organization_id,
+            email=author_email,
+            defaults={
+                'name': commit['commit']['author']['name'][:128],
+            }
+        )[0]
+        if author.name != commit['commit']['author']['name']:
+            author.update(name=commit['commit']['author']['name'])
+
+    # We do not want to update here because the information that comes
+    # from the webhook is of higher quality
+    Commit.objects.create_or_update(
+        repository_id=repo.id,
+        key=commit['sha'],
+        organization_id=repo.organization_id,
+        values=dict(
```

Comment:
based on your comment, do you want this to be defaults? https://github.com/getsentry/sentry/blob/master/src/sentry/db/models/query.py#L58. also, if you don't want to update, is there a reason you're not just creating and handling the integrity error?